nulib/meadow
# 2374 remove yarn

# Summary 

Replace `yarn` with `npm` throughout the entire dev/test/build process

# Specific Changes in this PR
- add an ADR documenting the switch (and the deprecation of ADR 11)
- remove all `yarn.lock` files and replace them with `package-lock.json`
- update GitHub Actions build
- update `Dockerfile`
- update EC2 instance install script
- update `README.md` to change development setup instructions

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

```
rm -rf deps _build assets/node_modules priv/nodejs/*/node_modules
```

Then run through the entire Meadow setup process, run the test suite, start the app, and make sure the UI and a small ingest seem to work OK.

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
